### PR TITLE
APIGOV-19048 - add healthchecking for condor connections

### PR DESCRIPTION
**Pull request by dfeldick**

### CODE REVIEW

**Review by kflan-io — APPROVED** [review]

**Comment by jcollins-axway** [comment]
I am confused, are you not registering the health check any longer? Just creating it as a job that is called health check and uses some health check constants?

**Comment by dfeldick** [comment]
Yes. Why would I need both a job and a registration of a healthchecker?

**Comment by sbolosan** [comment]
I think he's saying you need to register the actual job...

**Comment by dfeldick** [comment]
jobs.RegisterIntervalJob. Line 247.

**Comment by sbolosan** [comment]
hmmm...looks like you're doing that

File: pkg/traceability/traceability.go [hunk 1]
```diff
@@ -211,3 +230,78 @@ func updateEvent(batch publisher.Batch, events []publisher.Event) {
...
+			protocol: config.Protocol,
+			host:     config.Hosts[i],
+			proxyURL: config.Proxy.URL,
+			timeout:  config.Timeout,
+		}
+
+		hcJob := &condorHealthCheckJob{
+			agentHealthChecker: ta,
+		}
+		jobs.RegisterIntervalJob(hcJob, config.Timeout)
```
Comment (sbolosan) [inline pkg/traceability/traceability.go:247]: This can return an error.  You probably should check for that

Comment (dfeldick) [reply pkg/traceability/traceability.go:247]: I can and will. But if I dig into RegisterIntervalJob, it looks like it never really returns anything except nil.

**Comment by jcollins-axway** [comment]
> jobs.RegisterIntervalJob. Line 247.

That works. But then you should not run the function as if its a health check function with health check returns.  All that health check code could be in your job struct.  If you are not registering a healthcheck at all then don't use the healthcheck library at all, is wha tI mean.

**Comment by dfeldick** [comment]
I apparently mistakenly patterned my implementation after pollUsagesPlans in aws, where it does almost the exact same thing I did. However, I see what you are saying I think. I can move the healthcheck stuff from traceability.go directly into my job file and tweek it to just return success/fail or whatever, thereby eliminating any of the hc calls and vars.

**Comment by jcollins-axway** [comment]
> I apparently mistakenly patterned my implementation after pollUsagesPlans in aws, where it does almost the exact same thing I did. However, I see what you are saying I think. I can move the healthcheck stuff from traceability.go directly into my job file and tweek it to just return success/fail or whatever, thereby eliminating any of the hc calls and vars.

Those ones need updated.

File: pkg/traceability/traceability.go [hunk 2]
```diff
@@ -211,3 +227,82 @@ func updateEvent(batch publisher.Batch, events []publisher.Event) {
...
+		}
+		_, err := jobs.RegisterIntervalJob(hcJob, config.Timeout)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// func (ta *traceabilityAgentHealthChecker) healthcheck(name string) *hc.Status {
```
Comment (jcollins-axway) [inline pkg/traceability/traceability.go:252]: clean these comments up?

**Review by jcollins-axway — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

File: pkg/traceability/traceability.go (hunk 2 above)
Comment (dfeldick) [reply pkg/traceability/traceability.go:252]: Oops. Done.

**Pull request merged**